packages/sdk/src/utils.ts

```diff
@@ -182,13 +182,20 @@ const generateShareFilename = /* @__PURE__ */ (
 
 const getResourceUrl = (module: ModuleInfo, sourceUrl: string): string => {
   if ('getPublicPath' in module) {
-    const publicPath = new Function(module.getPublicPath)();
+    let publicPath;
+
+    if (!module.getPublicPath.startsWith('function')) {
+      publicPath = new Function(module.getPublicPath)();
+    } else {
+      publicPath = new Function('return ' + module.getPublicPath)()();
+    }
+
     return `${publicPath}${sourceUrl}`;
   } else if ('publicPath' in module) {
     return `${module.publicPath}${sourceUrl}`;
```

Comment:
I am facing a bug with my rollup implementation here.

If I pass the publicPath as 'auto' in the mf-manifest.json, there's no logic here to understand that 'auto' is a special string and call [inferAutoPublicPath](https://github.com/module-federation/core/blob/52bb94cd9566bcd8a5e61da8552e71b548729ff4/packages/sdk/src/generateSnapshotFromManifest.ts#L51)

It is currently a blocker for make my rollup federation plugin work without this

@ScriptedAlchemy Can you add it in the same PR or should I send a PR to fix it ?

Reply:
Here's an example where it doesn't work: https://deploy-preview-59--rollup-plugin-module-federation.netlify.app/packages/examples/project-a/dist/rollup/esm/

Click on the buttons from left to right ^ and you will see the following error:
<img width="1463" alt="Screenshot 2024-06-21 at 4 55 09 PM" src="https://github.com/module-federation/core/assets/6977429/0c8412b5-776c-494b-88f9-81c911e09a9f">

Reply:
if getPublicPath exists, you should delete publicPath - both cannot exist at once

Reply:
I don't have `getPublicPath`

The publicPath: auto doesn't work. 

Check the implementation of [getResourceUrl](https://github.com/module-federation/core/blob/52bb94cd9566bcd8a5e61da8552e71b548729ff4/packages/sdk/src/utils.ts#L183). It doesn't do anything for the case when the publicPath is set to the string "auto"